UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/Services/BuilderService.cs

```diff
@@ -1,22 +1,222 @@
+            this.fssService = fssService ?? throw new ArgumentNullException(nameof(fssService));
+            this.fileSystemHelper = fileSystemHelper ?? throw new ArgumentNullException(nameof(fileSystemHelper));
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            homeDirectoryPath = Path.Combine(configuration["HOME"], configuration["BespokeFolderName"]);
         }
 
-        public async Task<string> CreateBespokeExchangeSet(ConfigQueueMessage message)
+        public async Task<string> CreateBespokeExchangeSetAsync(ConfigQueueMessage configQueueMessage)
         {
-            await essService.PostProductIdentifiersData(message.EncCellNames.ToList(), message.ExchangeSetStandard);
+            string essBatchId = await RequestExchangeSetAsync(configQueueMessage);
+            (string essFileDownloadPath, List<FssBatchFile> essFiles) = await DownloadEssExchangeSetAsync(essBatchId);
+
+            ExtractExchangeSetZip(essFiles, essFileDownloadPath);
+
+            //Temporary Upload Code
+            #region TemporaryUploadCode
+            CreateZipFile(essFiles, essFileDownloadPath);
+
+            bool isBatchCreated = CreateBessBatchAsync(essFileDownloadPath, BessBatchFileExtension, Batch.BesBaseZipBatch).Result;
+
+            // temporary logs
+            if (isBatchCreated)
+            {
+                logger.LogInformation(EventIds.CreateBatchCompleted.ToEventId(), "Bess Base batch created {DateTime} | {CorrelationId}", DateTime.UtcNow, configQueueMessage.CorrelationId);
+            }
+            else
+            {
+                logger.LogError(EventIds.CreateBatchFailed.ToEventId(), "Bess Base batch failed {DateTime} | {CorrelationId}", DateTime.UtcNow, configQueueMessage.CorrelationId);
+            }
+
+            #endregion TemporaryUploadCode
 
             return "Exchange Set Created Successfully";
         }
+
+        private async Task<string> RequestExchangeSetAsync(ConfigQueueMessage configQueueMessage)
+        {
+            ExchangeSetResponseModel exchangeSetResponseModel = new();
+            if (configQueueMessage.Type == BessType.BASE.ToString())
+                exchangeSetResponseModel = await essService.PostProductIdentifiersData((List<string>)configQueueMessage.EncCellNames, configQueueMessage.ExchangeSetStandard);
+            else if (configQueueMessage.Type == BessType.UPDATE.ToString() ||
+                     configQueueMessage.Type == BessType.CHANGE.ToString())
+            {
+                ProductVersionsRequest productVersionsRequest = GetProductVersionDetails(configQueueMessage.EncCellNames);
+                exchangeSetResponseModel = await essService.GetProductDataProductVersions(productVersionsRequest, configQueueMessage.ExchangeSetStandard);
+            }
+            return CommonHelper.ExtractBatchId(exchangeSetResponseModel.Links.ExchangeSetBatchDetailsUri.Href);
+        }
```

Comment:
this this do any validation of what the initial list of cells which were requested to the cells fetched from the essService